index.js

```diff
@@ -65,6 +65,15 @@ class Transaction {
     }
 
     data = data || {}
+    // Ensure all transaction properties that are strings are 0x prefixed
+    if (Array.isArray(data) || typeof data === 'object') {
+      Object.keys(data).forEach(key => {
+        data[key] = typeof data[key] === 'string'
```

Comment:
Wouldn't `nonce` and `gasLimit` evaluate to false and not be prefixed since `typeof` 0 is a `number`?

But tests is passing... I am assuming `.serialize` turn everything to a string. Is this an expected behavior? Or should be also check for type `number` explicitly?

Reply:
The issue is not so much whether they are prefixed, as whether `defineProperties` of `ethereumjs-util` treats them as prefixed. It ultimately converts all values to buffers, but doesn't use the correct encoding when converting strings unless they are correctly hex prefixed. Numbers are correctly converted to buffers, as `defineProperties` handles them separately from strings.

Reply:
👍